worlds/spire/__init__.py

```diff
@@ -32,14 +32,15 @@ class SpireWorld(World):
     topology_present = False
     data_version = 1
     web = SpireWeb()
+    required_client_version = (0, 3, 7)
 
     item_name_to_id = {name: data.code for name, data in item_table.items()}
     location_name_to_id = location_table
 
     def _get_slot_data(self):
         return {
             'seed': "".join(self.multiworld.slot_seeds[self.player].choice(string.ascii_letters) for i in range(16)),
-            'character': self.multiworld.character[self.player],
+            'character': self.multiworld.character[self.player].get_option_name,
```

Comment:
get_option_name is a method that needs to be called, as it stands you're embedding the method itself, which will fail transfer to client, as python method bytecode is not json serializable.

Reply:
huh.... thats odd because this does work and properly sends the character string.

Reply:
I printed it to double check and it was the method, at least for me.

Reply:
While this works `current_key` is slightly better since it's a property

Reply:
oh yeah.. at new line 80 it gets overwritten with option.value, shadowing that this happened at all. Should probably be fixed then.

Reply:
yea that was silly. fixed.

Reply:
this will still return an integer for the choices from the character selection... does the client parse those correctly?

Reply:
Yea the client converts this to a string then looks for the string of "1", "2", "3", or "4" and maps them to the appropriate character/random option. Failing this it then uses that string to search all titles of installed characters. Failing that it says you get ironclad.